var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -28,6 +28,20 @@ def __init__(self, spec):
                             url_name='mpi_oneapi')
         super(IntelOneapiMpi, self).__init__(spec)
 
+    def setup_dependent_package(self, module, dep_spec):
+        dir = join_path(self.prefix, 'mpi', 'latest', 'bin')
+        self.spec.mpicc  = join_path(dir, 'mpicc')
+        self.spec.mpicxx = join_path(dir, 'mpicxx')
+        self.spec.mpif77 = join_path(dir, 'mpif77')
+        self.spec.mpifc  = join_path(dir, 'mpifc')
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('MPICH_CC', spack_cc)
```

Comment:
Do the `intel-oneapi-mpi` MPI compiler wrappers use these `MPICH`-prefixed variables? I would have guessed they would use something like `I_MPI_CC` (which is set by `spack.build_systems.intel.mpi_setup_dependent_build_environment`) etc.

Note that if the `mpicc` wrappers provided by this package don't use such a variable, then you don't have to set it at all. But as mentioned in https://github.com/spack/spack/issues/3487#issuecomment-287564230, these wrappers might use such a variable to locate e.g. the underlying CC compiler that should be used.

Reply:
They use the MPICH prefixed variables. That may not have been true for pre-oneapi releases of Intel MPI.

```
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ which mpicc
/opt/intel/oneapi/mpi/2021.1.1//bin/mpicc
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ mpicc -show test.c
gcc 'test.c' -I'/opt/intel/oneapi/mpi/2021.1.1/include' -L'/opt/intel/oneapi/mpi/2021.1.1/lib/release' -L'/opt/intel/oneapi/mpi/2021\
.1.1/lib' -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker '/opt/intel/oneapi/mpi/2021.1.1/lib/release' -Xlinker -rpath -Xlinker\
 '/opt/intel/oneapi/mpi/2021.1.1/lib' -lmpifort -lmpi -lrt -lpthread -Wl,-z,now -Wl,-z,relro -Wl,-z,noexecstack -Xlinker --enable-ne\
w-dtags -ldl
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ MPICH_CC=icx mpicc -show test.c
icx 'test.c' -I'/opt/intel/oneapi/mpi/2021.1.1/include' -L'/opt/intel/oneapi/mpi/2021.1.1/lib/release' -L'/opt/intel/oneapi/mpi/2021\
.1.1/lib' -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker '/opt/intel/oneapi/mpi/2021.1.1/lib/release' -Xlinker -rpath -Xlinker\
 '/opt/intel/oneapi/mpi/2021.1.1/lib' -lmpifort -lmpi -ldl -lrt -lpthread
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$
```